ibis/backends/dask/__init__.py

```diff
@@ -170,7 +170,9 @@ def get_schema(self, table_name, *, database=None):
             schema = self.schemas[table_name]
         except KeyError:
             df = self.dictionary[table_name]
-            self.schemas[table_name] = schema = PandasData.infer_table(df.head(1))
+            if isinstance(df, dd.DataFrame):
+                df = df.compute()
+            self.schemas[table_name] = schema = PandasData.infer_table(df)
```

Comment:
The original code was breaking a bunch of tests because in one of the datasets we're using there is a struct column that looks like `[{}, None, {1: "a", 2: "b"}]`. We need all of the data in that column in order to infer the struct type there...